### PR TITLE
GoMod: Fix analyzing projects without deps

### PR DESCRIPTION
**Pull request by fviernau**
See individual commits.

This is part of #6615. For that issue to be fixed GoMod needs to work in case of:
- unused dependencies + used dependencies
- only unused dependencies
- no dependencies at all

### CODE REVIEW

File: analyzer/src/main/kotlin/managers/GoMod.kt [hunk 1]
```diff
@@ -212,7 +212,7 @@ class GoMod(
 
     private fun ModuleInfo.toId(): Identifier =
         Identifier(
-            type = managerName.takeIf { version.isBlank() } ?: "Go",
+            type = managerName.takeIf { main } ?: "Go",
```
Comment (sschuberth) [inline analyzer/src/main/kotlin/managers/GoMod.kt:216]: Commit message nits:
- "determining *a* value"
- "without *a* version" (twice)
- Probably a comma before "this change is of non-functional nature"

Comment (sschuberth) [reply analyzer/src/main/kotlin/managers/GoMod.kt:216]: Not strictly related, but how do you think about renaming the `main` and `indirect` boolean properties to `isMain` and `isIndirect` in another commit for clarity?

File: analyzer/src/funTest/assets/projects/synthetic/gomod-no-deps-expected-output.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+---
```
Comment (sschuberth) [inline analyzer/src/funTest/assets/projects/synthetic/gomod-no-deps-expected-output.yml:1]: Commit message nits:
- "is zero edges" -> "*are* zero edges"

File: analyzer/src/main/kotlin/managers/GoMod.kt [hunk 3]
```diff
@@ -170,7 +170,8 @@ class GoMod(
                 moduleInfo(moduleName).toId()
             }
 
-        var graph = Graph()
+        val mainModuleId = moduleInfoForModuleName.values.single { it.main }.toId()
+        var graph = Graph(mutableMapOf(mainModuleId to setOf()))
```
Comment (sschuberth) [inline analyzer/src/main/kotlin/managers/GoMod.kt:174]: Use `emptySet()`.

**Review by sschuberth — DISMISSED** [review]

**Comment by sschuberth** [comment]
How does this relate to https://github.com/oss-review-toolkit/ort/pull/6638? Is it an alternative / replacement, and does it fix the same issue? (No issue is being reference in the commit messages.)

File: analyzer/src/main/kotlin/managers/GoMod.kt (hunk 1 above)
Comment (fviernau) [reply analyzer/src/main/kotlin/managers/GoMod.kt:216]: The names are exactly the same as in `golang` upstream data structures. I find it a tad simpler if these are consistent, which is why I'd prefer to keep these names.

Comment (sschuberth) [reply analyzer/src/main/kotlin/managers/GoMod.kt:216]: > The names are exactly the same as in `golang` upstream data structures.

That's not true, as they already differ in capitalization.

**Comment by fviernau** [comment]
> How does this relate to #6638? Is it an alternative / replacement, and does it fix the same issue? (No issue is being reference in the commit messages.)

PR #6638 is in context of #6615 and this PR is also in context of #6615.
I'm planning to fix #6615, as I already put quite a bit effort into it. So, I'll file another PR
which might replace #6638.

File: analyzer/src/main/kotlin/managers/GoMod.kt (hunk 1 above)
Comment (fviernau) [reply analyzer/src/main/kotlin/managers/GoMod.kt:216]: I think you got my point.

**Review by oheger-bosch — APPROVED** [review]

**Pull request merged**